src/data/visualizations/VisualizationsD2Repository.ts

```diff
@@ -0,0 +1,37 @@
+import { Future } from "../../domain/common/entities/Future";
+import { Id } from "../../domain/common/entities/Schemas";
+import { Instance } from "../../domain/instance/entities/Instance";
+import { Visualization } from "../../domain/visualizations/entities/Visualization";
+import {
+    GetVisualizationsOptions,
+    VisualizationsRepository,
+} from "../../domain/visualizations/repositories/VisualizationsRepository";
+import { D2Api } from "../../types/d2-api";
+import { getD2APiFromInstance } from "../../utils/d2-utils";
+import { apiToFuture } from "../common/utils/api-futures";
+
+export class VisualizationsD2Repository implements VisualizationsRepository {
+    private api: D2Api;
+
+    constructor(instance: Instance) {
+        this.api = getD2APiFromInstance(instance);
+    }
+
+    async getByIds(ids: Id[], options: GetVisualizationsOptions): Promise<Visualization[]> {
+        const fields = options.onlyRows ? "id,rows[dimension,items[id,name]]" : "*";
+
+        const $reqVisualizations = ids.map(id => {
+            return apiToFuture(
+                this.api.get<Visualization>(`/visualizations/${id}`, {
```

Comment:
That looks off. The option determines if we return all the fields or only id/rows. But the output type is Visualization, where all fields are available.

Not sure how will that fit in the current code of mdsync, but the general approach is to have two different methods. Here it's onlyRows: false used? if not, let's no implement unused code. (the other option would be conditional type, not recommended for our own code).

In summary, the specified return type of the method should be the real returned type.